libraries/ms-drivers/src/x86/pca9539r_gpio_expander.c

```diff
@@ -20,33 +19,38 @@ static I2CPort s_i2c_port = NUM_I2C_PORTS;
 static Pca9539rGpioSettings s_pin_settings[MAX_I2C_ADDRESSES][NUM_PCA9539R_GPIO_PINS];
 
 #ifdef MPXE
+
+static MxPca9539rStore s_stores[MAX_I2C_ADDRESSES];
+
 #define PCA9539_I2C_ADDRESS 0x74  // PCA9539 address used in the smoke test
 
-static void prv_export() {
+static void prv_export(void *key) {
+  int k = (intptr_t)(key);
   for (uint16_t j = 0; j < NUM_PCA9539R_GPIO_PINS; j++) {
-    s_store.state[j] = s_pin_settings[PCA9539_I2C_ADDRESS][j].state;
+    s_stores[k].state[j] = s_pin_settings[k][j].state;
   }
 
-  store_export(MX_STORE_TYPE__PCA9539R, &s_store, NULL);
+  store_export(MX_STORE_TYPE__PCA9539R, &s_stores[k], key);
 }
 
-static void update_store(ProtobufCBinaryData msg_buf, ProtobufCBinaryData mask_buf) {
+static void update_store(ProtobufCBinaryData msg_buf, ProtobufCBinaryData mask_buf, void *key) {
   MxPca9539rStore *msg = mx_pca9539r_store__unpack(NULL, msg_buf.len, msg_buf.data);
   MxPca9539rStore *mask = mx_pca9539r_store__unpack(NULL, mask_buf.len, mask_buf.data);
 
+  int k = (intptr_t)(key);
   for (uint16_t i = 0; i < mask->n_state; i++) {
     // only update state if mask is set
     if (mask->state[i] != 0) {
-      s_store.state[i] = msg->state[i];
-      if (s_pin_settings[PCA9539_I2C_ADDRESS][i].state != (uint8_t)msg->state[i]) {
-        s_pin_settings[PCA9539_I2C_ADDRESS][i].state = msg->state[i];
+      s_stores[k].state[i] = msg->state[i];
+      if (s_pin_settings[k][i].state != (uint)msg->state[i]) {
```

Comment:
does this actually compile? what's uint? shouldn't' it be uint8_t?

Reply:
It does compile. Yes you're right, it should be uint8_t. My bad.